packages/next-auth/src/next/index.ts

```diff
@@ -121,18 +120,25 @@ export async function unstable_getServerSession(
     experimentalRSCWarningShown = true
   }
 
-  const [req, res, options] = isRSC
-    ? [
-        {
-          headers: headers(),
-          cookies: nextCookies()
-            .getAll()
-            .reduce((acc, c) => ({ ...acc, [c.name]: c.value }), {}),
-        } as any,
-        { getHeader() {}, setCookie() {}, setHeader() {} } as any,
-        args[0],
-      ]
-    : args
+  let req, res, options: NextAuthOptions
+  if (isRSC) {
+    options = args[0]
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { headers, cookies } = require("next/headers")
+    req = {
+      headers,
+      cookies: Object.fromEntries(
+        cookies()
+          .getAll()
+          .map((c) => [c.name, c.value])
+      ),
+    }
+    res = { getHeader() {}, setCookie() {}, setHeader() {} }
+  } else {
+    req = args[0]
+    res = args[1]
+    options = args[2]
```

Comment:
Maybe it is worth to rewrite as
```js
[req, res, options] = args
```
for better readability?

Reply:
Syntactical preference. This is internal code anyway, but thanks for the suggestion. :+1:

Would actually be formatted to `;[req, res, options] = args` via Prettier, which I like less. :sweat_smile: